### PR TITLE
feat!: final v9 touchups

### PR DESCRIPTION
**Pull request by kanadgupta**
## 🧰 Changes

a few final touchups before we ship v9 🚀 
- [x] adds a deprecation notice to `rdme openapi`
- [x] removes the usage section from our docs in favor of a quick start
- [x] improves the documentation for our `--key` and `--version` flags
- [x] removes all `v10` migration guide language for now and adds a placeholder. i have a separate PR ([#1107](https://github.com/readmeio/rdme/pull/1107)) that adds that language back when we're ready. i figured i'd chunk this out so we can ship v9 without having to finalize the new `rdme openapi` command replacement.

## 🧬 QA & Testing

Do these doc changes look sound?

--- 

BREAKING CHANGE: `rdme openapi` is deprecated and will be replaced in `rdme@10` by a command with a simpler flag setup based on community feedback.

### CODE REVIEW

**Review by erunion — APPROVED** [review]

File: documentation/commands/categories.md
Comment (kanadgupta) [inline documentation/commands/categories.md:1]: the diff in all of these autogenerated `documentation/commands` files is the same — basically all of the changes in these files stem from the updates i made to `src/lib/flags.ts`!

File: documentation/migration-guide.md [hunk 1]
```diff
@@ -28,60 +28,12 @@ npx markdown-toc documentation/migration-guide.md --maxdepth 2 --bullets="-" -i
...
 <!-- prettier-ignore-start -->
-> [!NOTE]
-> `rdme@10` has not been released yet. If you're using ReadMe Refactored, stay tuned!
+> [!WARNING]
+> `rdme@10` has not been released yet, so this section is still under construction.
+>
+> [Please verify that you're looking at the latest docs for v9](https://github.com/readmeio/rdme/tree/v9/documentation/migration-guide.md). If you're using ReadMe Refactored, stay tuned!
 <!-- prettier-ignore-end -->
 
-### Overview
```
Comment (kanadgupta) [inline documentation/migration-guide.md:35]: removing all of this for now and bringing it back in #1107

**Review by kanadgupta — COMMENTED** [review]
@erunion @mjcuva sorry this needs another review 🫣 basically i went through our main README again and found that the whole usage section didn't make sense structurally so i reworked that in https://github.com/readmeio/rdme/pull/1106/commits/826fecaf957595f420a6ee319c54a183e9c5f40a and https://github.com/readmeio/rdme/pull/1106/commits/3a007da325323674ced4fc5a79ffa34a46b488d0.[](url)

File: README.md [hunk 2]
```diff
@@ -42,13 +42,39 @@ npm run build && npm run build:docs
...
+Validate an OpenAPI file in your working directory or any subdirectories ([see here for all command topics](#command-topics)):
+
+```sh
+rdme openapi validate
+```
+
+Every command has a help page, which you can access in [our docs](./documentation/commands) or via the CLI:
+
+```sh
+rdme openapi validate --help
```
Comment (erunion) [inline README.md:69]: v9 has space separated commands? i thought we were doing that starting in v10?

**Review by erunion — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (kanadgupta) [reply README.md:69]: oh nope sorry i did that in v9 just so it'd be easy to maintain our v9 and v10 documentation in parallel — any qualms with that?

Comment (erunion) [reply README.md:69]: nope was just confused

**Pull request merged**